ADALiOS/ADALiOS/ADAuthenticationRequest+AcquireAssertion.m

```diff
@@ -0,0 +1,232 @@
+// Copyright © Microsoft Open Technologies, Inc.
+//
+// All Rights Reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+// ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+// PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache License, Version 2.0 for the specific language
+// governing permissions and limitations under the License.
+
+#import "ADAuthenticationContext+Internal.h"
+#import "ADInstanceDiscovery.h"
+#import "ADUserIdentifier.h"
+#import "ADAuthenticationRequest.h"
+
+@implementation ADAuthenticationRequest (AcquireAssertion)
+
+- (void)acquireTokenForAssertion:(NSString*)samlAssertion
+                   assertionType:(ADAssertionType)assertionType
+                 completionBlock:(ADAuthenticationCallback)completionBlock
+{
+    
+    THROW_ON_NIL_ARGUMENT(completionBlock);
+    AD_REQUEST_CHECK_PROPERTY(_resource);
```

Comment:
Why do you  check the resource here? Resource is not in the parameter list. It should be checked when you created this request.